From afc1d7344408688b18748cff78bba8be9000ff36 Mon Sep 17 00:00:00 2001
From: Michael Glembotzki <Michael.Glembotzki@iris-sensing.com>
Date: Wed, 30 Mar 2022 15:18:40 +0200
Subject: [PATCH] imx8mp-irma6r2.dts: Enable i2c port expander pca9554 on
 interface unit

currently disabled to avoid breaking the PA

[    2.226092] pca953x 3-0021: using no AI

root@imx8mpevk:~# cd /sys/class/gpio/
root@imx8mpevk:/sys/class/gpio# echo 504 > export
root@imx8mpevk:/sys/class/gpio# echo out > gpio504/direction
root@imx8mpevk:/sys/class/gpio# echo 1 > gpio504/value

Signed-off-by: Michael Glembotzki <Michael.Glembotzki@iris-sensing.com>
Signed-off-by: Jasper Orschulko <jasper@fancydomain.eu>
---
 .../boot/dts/freescale/imx8mp-irma6r2.dts     | 26 +++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/imx8mp-irma6r2.dts b/arch/arm64/boot/dts/freescale/imx8mp-irma6r2.dts
index 679baf7b326b..593459d7c531 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-irma6r2.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mp-irma6r2.dts
@@ -210,6 +210,14 @@ &i2c1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_i2c1>;
 	status = "okay";
+
+	gpio@23 {
+		compatible = "nxp,pca9554";
+		reg = <0x23>;
+		gpio-controller;
+		#gpio-cells = <2>;
+		status = "disabled";
+	};
 };
 
 &i2c2 {
@@ -408,6 +416,18 @@ &i2c4 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_i2c4>;
 	status = "okay";
+
+	gpio@21 {
+		compatible = "nxp,pca9554";
+		reg = <0x21>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_pca9554>;
+		gpio-controller;
+		#gpio-cells = <2>;
+		interrupt-parent = <&gpio4>;
+		interrupts = <20 IRQ_TYPE_LEVEL_LOW>;
+		status = "disabled";
+	};
 };
 
 &uart1 { /* BT */
@@ -780,6 +800,12 @@ MX8MP_IOMUXC_SAI1_TXD5__GPIO4_IO17		0x10
 		>;
 	};
 
+	pinctrl_pca9554: pca9554_grp {
+		fsl,pins = <
+			MX8MP_IOMUXC_SAI1_MCLK__GPIO4_IO20		0x10
+		>;
+	};
+
 	pinctrl_csi_mclk: csi_mclk_grp {
 		fsl,pins = <
 			MX8MP_IOMUXC_GPIO1_IO15__CCM_CLKO2	0x50
-- 
2.46.0

